src/media_cache.rs

```diff
@@ -44,21 +51,41 @@ impl MediaCache {
     /// It will also optionally send updates to the given timeline update sender
     /// when a media request has completed.
     pub const fn new(
-        default_format: MediaFormatConst,
         timeline_update_sender: Option<crossbeam_channel::Sender<TimelineUpdate>>,
     ) -> Self {
         Self {
             cache: BTreeMap::new(),
-            default_format,
             timeline_update_sender,
         }
     }
 
     /// Gets media from the cache without sending a fetch request if the media is absent.
     ///
     /// This is suitable for use in a latency-sensitive context, such as a UI draw routine.
-    pub fn try_get_media(&self, mxc_uri: &OwnedMxcUri) -> Option<MediaCacheEntry> {
-        self.get(mxc_uri).map(|v| v.lock().unwrap().deref().clone())
+    pub fn try_get_media(&self, mxc_uri: &OwnedMxcUri, prefer_thumbnail: bool) -> MediaCacheEntry {
+        if let Some(v) = self.get(mxc_uri) {
+            for entry_and_format in v {
+                let mutex_guard = entry_and_format.lock().unwrap();
+                let entry_and_format = mutex_guard.deref().clone();
+
+                if prefer_thumbnail && !matches!(&entry_and_format.format, &MediaFormat::File)
```

Comment:
i think this is a typo, shouldn't it be:
```suggestion
                if prefer_thumbnail && !matches!(&entry_and_format.format, &MediaFormat::Thumbnail)
```

Reply:
this logic is correct, `! matches!(&entry_and_format.format, &MediaFormat::File)` is equal to `matches!(&entry_and_format.format, &MediaFormat::Thumbnail)`

Reply:
You know, `matches!(&entry_and_format.format, &MediaFormat::Thumbnail)` is just a little long, so i write that.